tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/RegionalAuthIntegrationTests.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Net;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Test.Common;
+using Microsoft.Identity.Test.Integration.net45.Infrastructure;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.Identity.Test.Unit;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Integration.HeadlessTests
+{
+    [TestClass]
+    public class RegionalAuthIntegrationTests
+    {
+        private static readonly string[] s_keyvaultScope = { "https://vault.azure.net/.default" };
+
+        private const string PublicCloudConfidentialClientID = "16dab2ba-145d-4b1b-8569-bf4b9aed4dc8";
+        private const string PublicCloudTestAuthority = "https://login.windows.net/72f988bf-86f1-41af-91ab-2d7cd011db47";
+        private KeyVaultSecretsProvider _keyVault;
+        private Dictionary<string, string> Dict = new Dictionary<string, string>
+        {
+            ["allowestsrnonmsi"] = "true"
+        };
+
+        [ClassInitialize]
+        public static void ClassInitialize(TestContext context)
+        {
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+        }
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            if (_keyVault == null)
```

Comment:
Make sure you clear any caches that may interfere with tests for this file. See other test classes on how to do this.

Reply:
Yeah I added that code in the cleanup method while I was trying it out. Will add it back here.